public/static/docs/command-reference/diff.md

```diff
@@ -1,41 +1,52 @@
 # diff
 
-Show changes between commits in the <abbr>DVC repository</abbr>, or between a
-commit and the <abbr>workspace</abbr>. The comparison can be narrowed down to
-specific target files/directories tracked by DVC.
+Show added, modified, or deleted DVC-tracked files and directories between
+commits in the <abbr>DVC repository</abbr>, or between a commit and the
+workspace.
 
 ## Synopsis
 
 ```usage
-usage: dvc diff [-h] [-q | -v] [-t TARGET] a_ref [b_ref]
+usage: dvc diff [-h] [-q | -v]
+                [--show-json] [--show-hash]
+                [a_rev] [b_rev]
 
 positional arguments:
-  a_rev                 Old Git commit to compare (defaults to HEAD)
-  b_rev                 New Git commit to compare (defaults to the
-                        current workspace)
+  a_rev      Old Git commit to compare (defaults to HEAD)
+  b_rev      New Git commit to compare (defaults to the current workspace)
 ```
 
 ## Description
 
-Given two commit hashes, branch or tag names, etc.
-([references](https://git-scm.com/docs/revisions)) `a_ref` and `b_ref`, this
-command shows a comparative summary of basic statistics related to files tracked
-by DVC: how many files were deleted/changed, and the file size differences.
+Output files and directories added, modified, deleted in a Git commit `b_rev` as
+compared to another Git commit `a_rev`. Both `a_rev` and `b_rev` accept any
+[Git revision]("https://git-scm.com/docs/gitrevisions") - branch or tag name,
+Git commit hash, etc.
 
-> Note that `dvc diff` does not show the line-to-line comparisons like
-> `git diff` or [GNU `diff`](https://www.gnu.org/software/diffutils/) can. This
-> is because the data data tracked by DVC comes in many formats such as
-> structured text, binary blobs, etc. For an example on how to create
-> line-to-line text file comparison, refer to
-> [issue #770](https://github.com/iterative/dvc/issues/770#issuecomment-512693256).
+It defaults to comparing the current workspace and the last commit (`HEAD`), if
+arguments `a_rev` and `b_rev` are not specified.
+
+Options `--show-hash` and `--show-json` can be used to modify format and details
+of the output produced. See the details and example below.
 
 `dvc diff` does not have an effect when the repository is not tracked by Git,
 for example when `dvc init` was used with the `--no-scm` option.
```

Comment:
> `dvc diff` does not have an effect... when `dvc init` was used with the `--no-scm` option

Somewhat unrelated but actually, I checked and this isn't correct. You can create a git repo, then a `dvc init --no-scm` in it, `dvc add` a data file, commit (A); change and `dvc add` the data file again, commit (B), and `dvc diff HEAD^` will work.

Is this buggy behavior? If not, we just need to rewrite the paragraph above to something more correct.

Reply:
it feels more or less correct to me - the main thing here is `when the repository is not tracked by Git`. In your example it is tracked by Git, right? unless I'm missing something.

Reply:
Yes it's tracked by Git, but I initialized the DVC project with `dvc init --no-scm`. The paragraph above reads ``...`dvc diff` does not have an effect... for example when `dvc init` was used with the `--no-scm` option`` but it does have an effect, as long as there is a Git repo underneath. Maybe just remove the part about `--no-scm` in the paragraph?

Reply:
But this makes me wonder whether `dvc init --no-scm` should actually check that there's no `git/` dir or something – totally unrelated now 😬

Reply:
Turns out there's an issue for this: #2472.

Reply:
It's fine, I still don't see a problem. It's intuitively clear what does it mean, even if implementation is not 100% correct and allows to have a mix of Git and --no-scm.

Reply:
Hmmm OK but just removing that last incorrect statement wouldn't hurt either.